internal/cli/federatedauthentication/identityprovider/list.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package identityprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115010/admin"
+)
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	*cli.ListOpts
+	store                store.IdentityProviderLister
+	federationSettingsID string
+	itemsPerPage         int
+	pageNum              int
+	idpType              string
+	protocol             string
+}
+
+const oidc = "OIDC"
+
+const listTemplate = `ID	DISPLAY NAME	ISSUER URI	CLIENT ID	IDP TYPE{{range valueOrEmptySlice .Results}}
+{{.Id}}	{{.DisplayName}}	{{.IssuerUri}}	{{.ClientId}}	{{.IdpType}}{{end}}
+`
+
+func (opts *ListOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *ListOpts) Run() error {
+	protocol := []string{opts.protocol}
+	idpType := []string{opts.idpType}
+	params := &atlasv2.ListIdentityProvidersApiParams{
+		FederationSettingsId: opts.federationSettingsID,
+		ItemsPerPage:         &opts.itemsPerPage,
+		PageNum:              &opts.pageNum,
+		Protocol:             &protocol,
+		IdpType:              &idpType,
```

Comment:
I don't see it on the tests but what happens if these are empty strings? I remember the SDK has some odd behaviour with empty string not being the same than nil so making sure we are handling it correctly

Reply:
can we use the default values of the flags instead?

Reply:
should we make flag not-required and if saml throw an error then?

Reply:
(we don't have the protocol flag until now as we're only supporting oidc)

Reply:
done that! and added support to saml since it works

Reply:
yeah if it works no harm to let it be